### PR TITLE
Refactored state commitment to `trie.go`

### PR DESCRIPTION
**Pull request by lunfardo314**
Type of change: breaking change, enhancement. Code changes are pretty local 
* Expected up to 30% less storage usage and faster DB updates
* All non-cluster tests pass.
* **Do not merge until all cluster tests checked**

Description:
* deleted package `trie` and related from the Wasp repo
* now state commitments are using `iotaledger/trie.go` depedency
* configuration of the commitment scheme is:
     * `blake2b` 160 bit hashing
     * 16-ary (hexary) trie 
     * terminal commitment optimization for values longer than 64 bytes
* added commitment to the anchor transaction essence into the `BlockInfo` structure. It enables fraud detection and proofs

### CODE REVIEW

**Review by dessaya — APPROVED** [review]

File: packages/chain/util.go [hunk 1]
```diff
@@ -6,23 +6,23 @@ import (
...
+		log.Debugf("STATE TRANSITION. Root commitment: %s", state.RootCommitment(msg.VirtualState.TrieNodeStore()))
 	} else {
 		log.Infof("ORIGIN STATE SAVED. State output id: %s", iscp.OID(msg.ChainOutput.ID()))
-		log.Debugf("ORIGIN STATE SAVED. state hash: %s", trie.RootCommitment(msg.VirtualState.TrieNodeStore()))
+		log.Debugf("ORIGIN STATE SAVED. state hash: %s", state.RootCommitment(msg.VirtualState.TrieNodeStore()))
 	}
 }
 
-// LogGovernanceTransition
+// LogGovernanceTransition LogGovernanceTransition
```
Comment (jorgemmsilva) [inline packages/chain/util.go]:
```suggestion
```

File: packages/state/loadsave.go [hunk 2]
```diff
@@ -8,34 +8,53 @@ import (
...
+func newTrieBatch(batch kvstore.BatchedMutations) mustTrieBatch {
+	return mustTrieBatch{batch}
+}
+
+func (b mustTrieBatch) Set(key, value []byte) {
+	k := dbkeys.MakeKey(dbkeys.ObjectTypeTrie, key)
+	if err := b.BatchedMutations.Set(k, value); err != nil {
 		panic(err)
 	}
 }
```
Comment (jorgemmsilva) [inline packages/state/loadsave.go]: I find this a bit confusing. `mustValueBatch` and `mustTrieBatch` do the same thing except how the key is built?
why can't we just use the existing `mustKVStoreBatch` with different prefixes?

btw, maybe we can rename these, why the `must` prefix? :sweat_smile:

File: packages/vm/core/blocklog/blockinfo.go [hunk 3]
```diff
@@ -14,18 +14,34 @@ import (
...
-	GasFeeCharged          uint64
+	BlockIndex                uint32 // not persistent. Set from key
+	Timestamp                 time.Time
+	TotalRequests             uint16
+	NumSuccessfulRequests     uint16 // which didn't panic
+	NumOffLedgerRequests      uint16
+	PreviousL1Commitment      state.L1Commitment     // always known
+	L1Commitment              *state.L1Commitment    // nil when not known yet for the current state
+	AnchorTransactionID       iotago.TransactionID   // of the input state
+	TransactionSubEssenceHash TransactionEssenceHash // always known even without state commitment. Needed for fraud proofs
```
Comment (jorgemmsilva) [inline packages/vm/core/blocklog/blockinfo.go:25]: what is a `TransactionSubEssenceHash` exactly? and how would it be used for fraud proofs? 
Why is it called "sub" hash?

File: tools/snapshot/main.go [hunk 4]
```diff
@@ -231,7 +231,7 @@ func createSnapshot(chainID *iscp.ChainID) {
 		fmt.Printf("error: %v\n", err)
 		os.Exit(1)
 	}
-	defer kvwriter.File.Close()
```
Comment (jorgemmsilva) [inline tools/snapshot/main.go:234]: why is this change needed?

**Review by jorgemmsilva — APPROVED** [review]

File: packages/state/loadsave.go (hunk 2 above)
Comment (lunfardo314) [reply packages/state/loadsave.go]: yes, redundant artefact due to refactoring. Will fix

Comment (lunfardo314) [reply packages/state/loadsave.go]: no, it is all fine!

File: packages/chain/util.go [hunk 5]
```diff
@@ -6,29 +6,22 @@ import (
...
 	}
 }
 
-// LogGovernanceTransition
-func LogGovernanceTransition(msg *ChainTransitionEventData, log *logger.Logger) {
-	stateHash, _ := hashing.HashValueFromBytes(msg.ChainOutput.GetStateMetadata())
-	log.Infof("GOVERNANCE TRANSITION state index #%d, anchor output: %s, state hash: %s",
-		msg.VirtualState.BlockIndex(), iscp.OID(msg.ChainOutput.ID()), stateHash.String())
-}
-
```
Comment (Juliusan) [inline packages/chain/util.go:31]: I used this function when implemented rotation.

**Review by Juliusan — APPROVED** [review]
